### DISCUSSION TITLE
Tests avec OpenGL

### DISCUSSION THREAD

**Discussion by BSoDium** [photonsquid/7map#8] [Ideas]
Salut à tous, je lache l'anglais deux minutes pour poser une question concernant la structure du repo : pour tester les fonctionnalités d'openGL (ou lWJGL pour être plus précis) on va avoir besoin de programmes "exemples", du coup est-ce que ce serait pas judicieux d'ajouter un dossier exemples_gl pour y mettre tout ça ? Je trouve juste dommage de tout laisser en local et de bosser chacun dans notre coin.
Dites moi ce que vous en pensez @7map/data @7map/rendering

**Comment by kingussopp** [comment]
Je trouve que c'est une bonne idée 👍

**Comment by seba1204** [comment]
Du coup, @Mmzhk21 et @kingussopp ont plus ou moins répondu à cette requête en créant une classe dédiée à cela.

Je me cependant demande si cette classe permet de faire n'importe quels tests unitaires (d'une classe quelconque), ou si on fait ces tests de manières plus conventionnelles dans le dossier `test` prévu à cet effet ?

**Reply by BSoDium** [reply]
Mais quoi qu'il en soit la classe de tests sur laquelle travaillent @Mmzhk21 et @kingussopp se trouvera dans le dossier `src/test` , ou du moins c'est ce qui est prévu non ?

**Reply by kingussopp** [reply]
@seba1204 je travaille avec la premiére methode parceque c'était la premiére idée que j'ai eu. Cependant, si vous préférez un dossier dédié ça m'ira

**Reply by seba1204** [reply]
> Mais quoi qu'il en soit la classe de tests sur laquelle travaillent @Mmzhk21 et @kingussopp se trouvera dans le dossier `src/test` , ou du moins c'est ce qui est prévu non ?

Oui, mais si on respecte la convention maven, il me semble qu'il faut reproduire exactement la même structure que le dossier `src/main`, mais pour les tests : il doit y avoir autant de fichiers *.java tests qu'il y a de *.java pas tests 😂 

c.f. [ce tuto](https://blog.engineering.publicissapient.fr/2014/01/13/separer-les-tests-dintegration-avec-maven/)

**Reply by seba1204** [reply]
> @seba1204 je travaille avec la premiére methode parceque c'était la premiére idée que j'ai eu. Cependant, si vous préférez un dossier dédié ça m'ira

La première méthode me convient, mais du coup cela veut dire que pour faire les tests de nos programmes, on va devoir modifier ta classe ?

**Reply by BSoDium** [reply]
Ce matin on s'est appelé avec @kingussopp et j'ai mis en place une structure similaire à ce dont tu parles, ie les tests sont dans le paquet `com.sevenmap` tout comme l'application et ses dépendances : 
```
📁 7map
...
┣ 📁 src
┃┣ 📁 main
┃┃ ┗ 📁 java
┃┃   ┗ 📁 com
┃┃     ┗ 📁 sevenmap
┃┗ 📁 test 
┃  ┗ 📁 java
┃    ┗ 📁 com
┃      ┗ 📁 sevenmap
...
```
Dis-moi ce que t'en penses 😁

**Reply by seba1204** [reply]
yes, du coup où seront les tests de :
```
📁 7map
...
┣ 📁 src
┃┣ 📁 main
┃┃ ┗ 📁 java
┃┃   ┗ 📁 com
┃┃     ┗ 📁 sevenmap
┃┃       ┗ 📁 data_handler
┃┃         ┗  📝 Feature.java
```
 ?

D'ailleurs, ce n'est pas le sujet de cette conversation, mais pourquoi on a changé de nom de package ? C'était pas 7map avant ?

**Reply by BSoDium** [reply]
logiquement sous `src/test/java/com/sevenmap/data_handler`
En ce qui concerne le changement de nom du package, je te renvoie à #29 ainsi qu'aux [conventions de nommage](https://docs.oracle.com/javase/specs/jls/se7/html/jls-3.html#jls-3.8) de paquets en java
